runtime/libffi/z/sysvz64.s

```diff
@@ -527,9 +884,141 @@ RLL      DS 0H
          B RET
 
 RV       DS 0H
-RS       DS 0H
          B RET
 
+RS       DS 0H
+         LG 7,(2176+(((DSASZ+31)/32)*32)+24)(,4)
+         LG 14,(2176+(((DSASZ+31)/32)*32)+8)(,4)
+         LG 14,0(,14)           ecif->cif
+         LG 9,16(,14)           cif->rtype
+         LG 14,0(,9)            rtype->size
+         CGIBH 14,24,RSP
+
+* now we need to determine how to return the struct
+* this label determines if the struct is <= 24 bytes
+* if it is we return using gprs/frprs
+* we then loop over rtype->elements
+* to determine whether this struct is all floating point
+* or if it is contains integral types
+* if we have just floating point types use FPRs as appropriate
+* but only if we have just the same type of float
+* a struct mixing float, double and long double will use GPRs
+* i have no idea why it is done this way, but it is
+* also if we use more than 2 floating point types we use GPRs
+* if we have at least one integral type, we use GPRs based on size
+* if the struct is > 24 bytes we have already passed in the return
+* pointer as the first argument to the callee "fn" 
+* so that work was outsource to the compiler
+* turning this into some simple cases for clarity
+* if struct <= 24 bytes 
+*   if elements contains >1 integral types -> GPRs
+*   if elements contains only one type of float
+*     if num elements <= 2 -> FPRs
+*     else -> GPRs
+         
+
+* currently we have 
+* r9 = cif->rtype
+* r7 = ecif->rvalue
+RSREG    DS 0H
+         LG 11,16(,9)         rtype->elements
+         LA 15,0              have we seen a float yet?
```

Comment:
This instruction is setting up the scratch registers for processing the return values. Comment seems to be out of place / may be need more clear comment.